docs/ops/sort/NonMaxSuppression_9.md

```diff
@@ -0,0 +1,134 @@
+
+**Versioned name**: *NonMaxSuppression-9*
+
+**Category**: *Sorting and maximization*
+
+**Short description**: *NonMaxSuppression* performs non maximum suppression of the boxes with predicted scores.
+
+**Detailed description**: *NonMaxSuppression* performs non maximum suppression algorithm as described below:
+
+1.  Let `B = [b_0,...,b_n]` be the list of initial detection boxes, `S = [s_0,...,s_N]` be  the list of corresponding scores.
+2.  Let `D = []` be an initial collection of resulting boxes.
+3.  If `B` is empty then go to step 8.
+4.  Take the box with highest score. Suppose that it is the box `b` with the score `s`.
+5.  Delete `b` from `B`.
+6.  If the score `s` is greater or equal than `score_threshold`  then add `b` to `D` else go to step 8.
+7.  For each input box `b_i` from `B` and the corresponding score `s_i`, set `s_i = s_i * func(IOU(b_i, b))` and go to step 3.
+8.  Return `D`, a collection of the corresponding scores `S`, and the number of elements in `D`.
+
+when `soft_nms_sigma == 0`, `func(iou) = 1 if iou <= iou_threshold else 0`, else `func(iou) = exp(-0.5 * iou * iou / soft_nms_sigma) if soft_NMS_suppressed_by_IOU == false`, else `func(iou) = exp(-0.5 * iou * iou / soft_nms_sigma) if iou <= iou_threshold else 0`.
+
+This algorithm is applied independently to each class of each batch element. The total number of output boxes for each
+class must not exceed `max_output_boxes_per_class`.
+
+**Attributes**:
+
+* *box_encoding*
+
+  * **Description**: *box_encoding* specifies the format of boxes data encoding.
+  * **Range of values**: "corner" or "center"
+    * *corner* - the box data is supplied as `[y1, x1, y2, x2]` where `(y1, x1)` and `(y2, x2)` are the coordinates of any diagonal pair of box corners.
+    * *center* - the box data is supplied as `[x_center, y_center, width, height]`.
+  * **Type**: string
+  * **Default value**: "corner"
+  * **Required**: *no*
+
+* *sort_result_descending*
+
+  * **Description**: *sort_result_descending* is a flag that specifies whenever it is necessary to sort selected boxes across batches or not.
+  * **Range of values**: true of false
+    * *true* - sort selected boxes across batches.
+    * *false* - do not sort selected boxes across batches (boxes are sorted per class).
+  * **Type**: boolean
+  * **Default value**: true
+  * **Required**: *no*
+
+* *output_type*
+
+  * **Description**: the output tensor type
+  * **Range of values**: "i64" or "i32"
+  * **Type**: string
+  * **Default value**: "i64"
+  * **Required**: *no*
+  
+* *soft_NMS_suppressed_by_IOU*
+
+  * **Description**: *soft_NMS_suppressed_by_IOU* is a flag that specifies whenever soft NMS is suppressed by IOU. It is used only when `soft_nms_sigma != 0`.
+  * **Range of values**: true of false
+    * *true* - soft NMS is suppressed by IOU.
+    * *false* - soft NMS is not suppressed by IOU.
+  * **Type**: boolean
+  * **Default value**: false
+  * **Required**: *no*
```

Comment:
Maybe this attribute should have default value `true`? If the default value is `false`, it can affect to models that have already been enabled, because all enabled models with NMS in IRs were converted with using NMS-5, and NMS-5 behaves as if `soft_NMS_suppressed_by_IOU == true`.

Reply:
As for this remark about the default value?